src/main/java/org/jabref/logic/importer/RawFetcher.java

```diff
@@ -0,0 +1,56 @@
+package org.jabref.logic.importer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.util.Collections;
+import java.util.List;
+
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.strings.StringUtil;
+
+public interface RawFetcher extends SearchBasedParserFetcher {
+
+    /**
+     * Send queries with arbitrary url parameters.
+     * <p>
+     * This is used to test the library APIs for their advanced search options e.g. fielded search.
+     *
+     * @param urlParameters The given URL parameters as string should start with "?"
+     * @return result of the query with the given URL parameters
+     */
+    default List<BibEntry> performRawSearch(String urlParameters) throws FetcherException, IllegalArgumentException {
+        if (StringUtil.isBlank(urlParameters)) {
+            return Collections.emptyList();
+        }
+        if (urlParameters.startsWith("?")) {
+            urlParameters = urlParameters.replaceFirst("\\?", "");
```

Comment:
In this case, following code is IMHO more reable

```suggestion
            urlParameters = urlParameters.substring(1);
```